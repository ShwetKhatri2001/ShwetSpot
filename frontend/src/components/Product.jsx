import React from 'react';
import { Link } from 'react-router-dom';
import { Card } from 'react-bootstrap';
import Rating from './Rating';

const Product = ({ product }) => {
  return (
    <Card className="mb-3 p-3 rounded h-auto" id="product">
      <Link to={`/product/${product._id}`}>
        <Card.Img src={product.image} variant="top" />
      </Link>

      <Card.Body>
        <Link to={`/product/${product._id}`}>
          <Card.Title as="div">
            <h4>{product.name}</h4>
          </Card.Title>
        </Link>

        {product.numReviews > 1 || product.numReviews === 0 ? (
          <Card.Text as="div">
            <Rating
              value={product.rating}
              text={`${product.numReviews} reviews`}
            />
          </Card.Text>
        ) : (
          <Card.Text as="div">
            <Rating
              value={product.rating}
              text={`${product.numReviews} review`}
            />
          </Card.Text>
        )}

        <Card.Text as="h4" className="pb-0 pt-2">
         &#8377; {product.price} 
        </Card.Text>
      </Card.Body>
    </Card>
  );
};

export default Product;
