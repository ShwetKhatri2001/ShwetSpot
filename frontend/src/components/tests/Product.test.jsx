import React from 'react';
import renderer from 'react-test-renderer';

import Product from '../Product';
import { MemoryRouter } from 'react-router-dom';

it('should render correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Product
          product={{
            _id: 1,
            image: '/location/images/cat.png',
            name: 'product',
            rating: 5,
            numReviews: 2,
          }}
        />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
