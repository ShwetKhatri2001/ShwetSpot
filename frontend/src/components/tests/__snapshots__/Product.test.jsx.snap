// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="mb-3 p-3 rounded h-auto card"
  id="product"
>
  <a
    href="/product/1"
    onClick={[Function]}
  >
    <img
      className="card-img-top"
      src="/location/images/cat.png"
    />
  </a>
  <div
    className="card-body"
  >
    <a
      href="/product/1"
      onClick={[Function]}
    >
      <div
        className="card-title"
      >
        <strong>
          product
        </strong>
      </div>
    </a>
    <div
      className="card-text"
    >
      <div
        className="rating"
      >
        <span>
          <i
            className="fas fa-star"
            style={
              Object {
                "color": "#4bbf73",
              }
            }
          />
        </span>
        <span>
          <i
            className="fas fa-star"
            style={
              Object {
                "color": "#4bbf73",
              }
            }
          />
        </span>
        <span>
          <i
            className="fas fa-star"
            style={
              Object {
                "color": "#4bbf73",
              }
            }
          />
        </span>
        <span>
          <i
            className="fas fa-star"
            style={
              Object {
                "color": "#4bbf73",
              }
            }
          />
        </span>
        <span>
          <i
            className="fas fa-star"
            style={
              Object {
                "color": "#4bbf73",
              }
            }
          />
        </span>
        <span>
          2 reviews
        </span>
      </div>
    </div>
    <h3
      className="pb-0 card-text"
    />
    <h5
      className="card-text"
    >
      credits
    </h5>
  </div>
</div>
`;
